File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.cpp
class Solution {
public:
    int minimumCardPickup(vector<int>& cards) {
        int n=cards.size();
        unordered_map<int,int>mp;
        int len=INT_MAX;
        for(int i=0;i<n;i++){
            if(mp.find(cards[i])!=mp.end()){
                len=min(len,i-mp[cards[i]]+1);
            }
            mp[cards[i]]=i;
        }
        return len==INT_MAX?-1:len;
    }
};

File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.js
/**
 * @param {number[]} cards
 * @return {number}
 */
var minimumCardPickup = function(cards) {
    const mp={};
    let len=Number.MAX_SAFE_INTEGER;
    for(let i=0;i<cards.length;i++){
        if(cards[i] in mp){
            if(len>(i-mp[cards[i]]+1)){
                len=i-mp[cards[i]]+1;
            }
            
        }
        mp[cards[i]]=i;
    }
    return len!=Number.MAX_SAFE_INTEGER?len:-1;
};